src/main/groovy/org/shipkit/internal/gradle/VersionUpgradeConsumerPlugin.java

```diff
@@ -9,170 +9,164 @@
+ *
+ *      versionUpgrade{
+ *          baseBranch = 'release/2.x'
+ *          buildFile = file('gradle.properties')
+ *      }
+ *
+ * and then call it:
+ *
+ * ./gradlew performVersionUpgrade -PdependencyNewVersion=org.shipkit:shipkit:1.2.3
+ *
  */
 public class VersionUpgradeConsumerPlugin implements Plugin<Project> {
 
     private static final Logger LOG = Logging.getLogger(VersionUpgradeConsumerPlugin.class);
 
-    public static final String VERSION_UPGRADE_CHECKOUT_BASE_BRANCH = "versionUpgradeCheckoutBaseBranch";
-    public static final String VERSION_UPGRADE_CHECKOUT_VERSION_BRANCH = "versionUpgradeCheckoutVersionBranch";
-    public static final String VERSION_UPGRADE_REPLACE_VERSION = "versionUpgradeReplaceVersion";
-    public static final String VERSION_UPGRADE_GIT_COMMIT = "versionUpgradeGitCommit";
-    public static final String VERSION_UPGRADE_GIT_PUSH = "versionUpgradeGitPush";
-    public static final String VERSION_UPGRADE_CREATE_PULL_REQUEST = "versionUpgradeCreatePullRequest";
+    public static final String CHECKOUT_BASE_BRANCH = "checkoutBaseBranch";
+    public static final String CHECKOUT_VERSION_BRANCH = "checkoutVersionBranch";
+    public static final String REPLACE_VERSION = "replaceVersion";
+    public static final String COMMIT_VERSION_UPGRADE = "commitVersionUpgrade";
+    public static final String PUSH_VERSION_UPGRADE = "pushVersionUpgrade";
+    public static final String CREATE_PULL_REQUEST = "createPullRequest";
     public static final String PERFORM_VERSION_UPGRADE = "performVersionUpgrade";
 
     public static final String DEPENDENCY_NEW_VERSION = "dependencyNewVersion";
-    public static final String DEPENDENCY_BUILD_FILE = "dependencyBuildFile";
-    public static final String DEPENDENCY_PATTERN = "dependencyPattern";
-    public static final String BASE_BRANCH = "baseBranch";
-
-    public static final String DEPENDENCY_PATTERN_DEFAULT = "org.shipkit:shipkit:{VERSION}";
-    public static final String BUILD_FILE_DEFAULT = "build.gradle";
-    public static final String BASE_BRANCH_DEFAULT = "master";
-
-
-    private EnvVariables envVariables;
 
-    @Inject
-    public VersionUpgradeConsumerPlugin(){
-        this(new EnvVariables());
-    }
-
-    @ExposedForTesting
-    public VersionUpgradeConsumerPlugin(EnvVariables envVariables){
-        this.envVariables = envVariables;
-    }
+    private VersionUpgrade versionUpgrade;
 
     @Override
     public void apply(final Project project) {
         LOG.lifecycle("Applying VersionUpgradeConsumerPlugin, beware that it's is INCUBATING state, so its API may change!");
         final ReleaseConfiguration conf = project.getPlugins().apply(ReleaseConfigurationPlugin.class).getConfiguration();
 
-        TaskMaker.task(project, VERSION_UPGRADE_CHECKOUT_BASE_BRANCH, GitCheckOutTask.class, new Action<GitCheckOutTask>() {
+        versionUpgrade = project.getExtensions().create("versionUpgrade", VersionUpgrade.class);
```

Comment:
Since more extensions are probably coming, what do think about creating some semantics / conventions around them. For example:
- plugin id "version-upgrade-customer", extension id "version-upgrade-customer", extension class name "VersionUpgradeCustomerExtension"

No need to change it at this point given that the plugin is incubating - food for thought. We would do the same with "shipkit" extension and rename "ReleaseConfiguration" to "ShipkitExtension".

Strong conventions will make the code consistent, easy to work with and easy to use by end users.